### ISSUE TITLE
Panel header accessibility improvements

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#1099]
I'd propose a few, simple, changes to the panel header (the sidebars header):

<img width="294" alt="screen shot 2017-06-10 at 11 38 21" src="https://user-images.githubusercontent.com/1682452/27001780-19777e9c-4dd2-11e7-9af9-e640cf4a50b6.png">

- the header label should be a `<h2>` heading, this is not just for abstract semantics but because headings empower screen reader users and allow them to quickly navigate content and jump from on section to another section using the screen reader built-in keyboard shortcuts
- add an `aria-label` to the the admin settings buttons
- add an `aria-label` to the the close button

Note: I'm not sure what the plans are for the admin settings button, I'm guessing it's meant to be a link to the WP settings screen. If so, it should be a link, not a button. Also not sure if it's going to be removed. For now I'm just going to add a label, I guess other changes can be done later.

Quick video to illustrate how such a simple change (the heading) can effectively improve content navigation for screen reader users. Without this, they would be forced to navigate through all content to find the sidebar or use other keyboard shortcuts with less predictable results:

https://cloudup.com/cNqOV4dpeLN
